components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/LoginEventHookHandler.java

```diff
@@ -0,0 +1,277 @@
+            handleLoginEvent(event, context, new ArrayList<>(Arrays.asList(failedStep)), false);
+        } else {
+            logDebug(log, String.format("Event %s received, but ignored as publishing config is disabled", event.getEventName()));
+        }
+    }
+
+    private void handleLoginEvent(Event event, AuthenticationContext context, List<AuthStep> authSteps, boolean isSuccess)
+            throws IdentityEventException {
+
+        AuthenticationData authenticationData;
+        if (isSuccess) {
+            authenticationData = AnalyticsLoginDataPublisherUtils.buildAuthnDataForAuthentication(event);
+        } else {
+            authenticationData = AnalyticsLoginDataPublisherUtils.buildAuthnDataForAuthnStep(event);
+        }
+
+        if (authenticationData.isPassive()) {
+            return;
+        }
+
+        try {
+            LoginEventPayloadBuilder payloadBuilder = PayloadBuilderFactory.getLoginEventPayloadBuilder(EVENT_SCHEMA_TYPE_WSO2);
+            AuthenticatedUser authenticatedUser = EventHookHandlerUtils.getAuthenticatedUserFromEvent(event);
+            EventHookHandlerUtils.setLocalUserClaims(authenticatedUser, context);
+            EventData eventData = new EventData(authenticationData, authenticatedUser, null, authSteps, context,
+                    context.getLoginTenantDomain());
+            EventPayload eventPayload = isSuccess ? payloadBuilder.buildAuthenticationSuccessEvent(eventData)
+                    : payloadBuilder.buildAuthenticationFailedEvent(eventData);
+
+            publishEventPayload(context, eventPayload, isSuccess);
+        } catch (Exception e) {
+            throw new IdentityEventException(String.format("Error while handling %s event.", event.getEventName()), e);
+        }
+    }
+
+    private void publishEventPayload(AuthenticationContext context, EventPayload eventPayload,
+                                     boolean isSuccess) throws Exception {
+
+        EventContext eventContext = new EventContext();
+        eventContext.setTenantDomain(context.getLoginTenantDomain());
+        String eventUri = EventHookHandlerUtils.getEventUri(isSuccess ? Constants.EventHandlerKey.LOGIN_SUCCESS_EVENT
+                : Constants.EventHandlerKey.LOGIN_FAILED_EVENT);
+        SecurityEventTokenPayload securityEventTokenPayload = EventHookHandlerUtils.buildSecurityEventToken(eventPayload,
+                context, eventUri, eventContext.getTenantDomain());
+        eventContext.setEventUri(eventUri);
+        EventHookHandlerDataHolder.getInstance().getEventPublisherService()
+                .publish(securityEventTokenPayload, eventContext);
+    }
+
+    private boolean isLoginEventHandlerEnabled() {
+
+        String enablePropertyKey = Constants.LOGIN_EVENT_HOOK_NAME + "." + Constants.ENABLE;
+        return this.configs != null && this.configs.getModuleProperties() != null &&
+                Boolean.parseBoolean(configs.getModuleProperties().getProperty(enablePropertyKey));
+    }
```

Comment:
As per the looks of this, event handler is enabled over a server level configuration.
In that case you should not check this at can handle.
Rather you can register the event handlers at startup based on this config

Reply:
fixed with https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1/commits/f196ef27141312ec28d2fb649831a766aecadbf8